iotdb-core/datanode/src/main/java/org/apache/iotdb/db/pipe/task/subtask/connector/PipeRealtimePriorityBlockingQueue.java

```diff
@@ -71,9 +78,20 @@ public boolean put(final Event event) {
 
   @Override
   public Event directPoll() {
-    Event event = super.directPoll();
+    Event event = null;
+    if (eventCount.get() >= pendingQueueConsumeThreshold && !tsfileInsertEventDeque.isEmpty()) {
+      event = tsfileInsertEventDeque.pollLast();
+      resetEventCount();
+    }
+
+    if (Objects.isNull(event)) {
+      event = super.directPoll();
+      eventCount.incrementAndGet();
+    }
+
     if (Objects.isNull(event)) {
       event = tsfileInsertEventDeque.pollLast();
+      resetEventCount();
```

Comment:
Do not reset here...

Reply:
Only poll first shall be considered as an effective commit and shall reset the counter. Poll last oppositely shall increase the count...